### PR TITLE
Add v1beta1 version

### PR DESCRIPTION
**Pull request by alebedev87**
What's inside:
- New version added in `api/v1beta1/`
- No conversions added as `v1beta1` is identical to `v1alpha1`
- `v1beta1` is the storage version now
- Webhook is moved to `api/v1beta1/`, it handles only `v1beta1` (`v1alpha1` objects are also served by API, see the notes)
- Code of the controllers is migrated to `v1beta1`
- Manifests updated: 
   - CRD got a new new version
   - Webhook updated with the new version
   - `v1beta1` samples added
- e2e tests migrated to `v1beta1`, one test is added for the check of `v1alpha1` resources

Notes:
- The controller and the validating webhook are setup to work with `v1beta1` only. This works out even for the objects created before the upgrade to this version of the operator. Because the version which is being watched/used is the [preferred](https://book.kubebuilder.io/multiversion-tutorial/deployment.html#kubectl-and-preferred-versions)(has to do with [the version priority](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#version-priority) too), so API will take care of making the conversion. This is different from where the objects are stored though. The objects created before the new version was added to CRD are still stored as the previous version until they are touched (patch/update).
- Migration to `stable` channel is done in a followup PR: https://github.com/openshift/external-dns-operator/pull/146. The reason is to not block other PR as this change needs the CI update (bundle installation).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -44,7 +44,7 @@ SETUP_ENVTEST := go run sigs.k8s.io/controller-runtime/tools/setup-envtest
 K8S_ENVTEST_VERSION := 1.21.4
 
 PACKAGE=github.com/openshift/external-dns-operator
-MAIN_PACKAGE=$(PACKAGE)/cmd/external-dns-operator
+MAIN_PACKAGE=$(PACKAGE)
```
Comment (DhritiShikhar) [inline Makefile:47]: `$(PACKAGE)` and `$(MAIN_PACKAGE)` has same value. Is there a reason to keep both of them?

File: api/v1beta1/externaldns_types.go [hunk 2]
```diff
@@ -0,0 +1,603 @@
...
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	Kind string `json:"kind"`
+
+	// Version is the API version
+	// of the given resource kind for
+	// ExternalDNS to use.
+	//
+	// e.g. "externaldns.k8s.io/v1alpha1"
```
Comment (DhritiShikhar) [inline api/v1beta1/externaldns_types.go:553]:
```suggestion
	// e.g. "externaldns.k8s.io/v1beta1"
```

File: config/samples/infoblox/operator_v1beta1_infoblox_openshift.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+apiVersion: externaldns.olm.openshift.io/v1beta1
```
Comment (DhritiShikhar) [inline config/samples/infoblox/operator_v1beta1_infoblox_openshift.yaml:1]: I see samples added for AWS, GCP, Azure and Infoblox but not Bluecat.. Is there a reason for that or just missed?

File: api/v1beta1/externaldns_types.go (hunk 2 above)
Comment (alebedev87) [reply api/v1beta1/externaldns_types.go:553]: It is the version of the [upstream CRD](https://github.com/kubernetes-sigs/external-dns/blob/master/docs/contributing/crd-source/crd-manifest.yaml#L20), not our ExternalDNS API.

File: Makefile (hunk 1 above)
Comment (alebedev87) [reply Makefile]: Not anymore, removed `MAIN_PACKAGE`

File: config/samples/infoblox/operator_v1beta1_infoblox_openshift.yaml (hunk 3 above)
Comment (alebedev87) [reply config/samples/infoblox/operator_v1beta1_infoblox_openshift.yaml:1]: Added BlueCat samples too

File: Makefile (hunk 1 above)
Comment (Miciah) [reply Makefile]: What was the reason for moving the main executable out of `cmd/`?  Is this change directly related somehow to the v1beta1 migration?

**Review by Miciah — COMMENTED** [review]
Looks good overall.  However, I'm wondering about the move of the main executable out of `cmd/`; is this change directly related somehow to the v1beta1 migration?

File: bundle/manifests/external-dns-operator.clusterserviceversion.yaml [hunk 4]
```diff
@@ -60,6 +60,36 @@ metadata:
...
+                }
+              },
+              "type": "BlueCat"
+            },
+            "source": {
+              "openshiftRouteOptions": {
+                "routerName": "default"
+              },
+              "type": "OpenShiftRoute"
+            }
```
[the remark this answers is not in the document]
Comment (alebedev87) [reply bundle/manifests/external-dns-operator.clusterserviceversion.yaml:90]: Some examples have `zones` field, some `domains` I thought of giving multiple different examples.
BlueCat example uses `domains` field to filter the zones by the domain name.

File: PROJECT [hunk 5]
```diff
@@ -15,4 +15,12 @@ resources:
...
     validation: true
     webhookVersion: v1
+- api:
+    crdVersion: v1
+    namespaced: true
+  domain: openshift.io
+  group: externaldns.olm
+  kind: ExternalDNS
+  path: github.com/openshift/external-dns-operator/api/v1beta1
+  version: v1beta1
```
[the remark this answers is not in the document]
Comment (alebedev87) [reply PROJECT:25]: Good question.
Looking at the [kubebuilder docs](https://book.kubebuilder.io/reference/project-config.html): `controller` and `webhooks` fields are added at the scaffolding time. Technically both the controller and the webhook were scaffolded for the version `v1alpha1`, so it reflects well the history behind this operator. Also, I didn't see any impact on the runtime.

**Comment by alebedev87** [comment]
@Miciah : no, the move of `main.go` out of `cmd/` is not related to the migration to `v1beta1`. It's the new kubebuilder project layout: before it was scaffolding into `cmd/`, now it's putting `main.go` in the root of the repository. I updated `operator-sdk` version meanwhile and the newer version could not scaffold the new api if `main.go` was not in the root.

**Pull request merged**